src/Paprika/Chain/Blockchain.cs

```diff
@@ -944,7 +946,20 @@ private void EnsureHash()
             }
         }
 
-        private BufferPool Pool => _blockchain._pool;
+        /// <summary>
+        /// Run merkle behaviour for storage tries only
+        /// </summary>
+        public void RecalculateStorageTries()
+        {
+            ((ComputeMerkleBehavior)_blockchain._preCommit).RecalculateStorageTries(this, _cacheBudgetPreCommit);
```

Comment:
Extract `RecalculateStorageTries` to the precommit interface so no cast is done here?

Reply:
Removed the method - unused